### ISSUE TITLE
Pyodide hangs after downloading

### ISSUE THREAD

**Issue by act65** [iodide-project/pyodide#387]
Trying to do https://alpha.iodide.io/notebooks/300/? but pyodide just hangs after downloading the plugin.

```
Loading Python language plugin
Python plugin downloaded, initializing
```

Am using Chrome 73 on ChromeOS 11647.

**Comment by osper66** [comment]
My iPad Pro's safari/chrome do same things.

**Comment by waveforms** [comment]
I had the same problem on a local install of iodide. Mac book /Chrome 73. The browser console gave multiple react related errors. What I did: compiled pyodide resources in docker. Copied all the resources over into iodide-master/build/pyodide-0.11.0 overwiting if the resource already existed. Ran the server version of iodide. Still a few errors but all the example code from https://alpha.iodide.io/notebooks/300/ works in my local instance. 

I am now looking for a way to require a password to go from "report" to "explore"

**Comment by osper66** [comment]
After  chromebook(using Chrome OS ver. 72.0.3626.122) was hanged on acessing to https://alpha.iodide.io/notebooks/300/? and executing this, this chrome's developer tool showed following message.

> Uncaught RangeError: WebAssembly.Memory(): could not allocate memory
>     at alpha.iodide.app/pyodide-0.10.0/pyodide.asm.js:8
>     at loadScript (eval at s.addEventListener (iodide.eval-frame.dev.js:93), <anonymous>:355:22)
>     at HTMLScriptElement.script.onload (eval at s.addEventListener (iodide.eval-frame.dev.js:93), <anonymous>:83:32)

 Is this wasm's execution out of memory error?

**Comment by act65** [comment]
Ok, the error I get is;

```
Uncaught RangeError: Invalid typed array length: undefined
    at new Int8Array (<anonymous>)
    at updateGlobalBufferViews (pyodide.asm.js:8)
    at pyodide.asm.js:8
    at eval (eval at <anonymous> (iodide.eval-frame.dev.js:93), <anonymous>:355:22)
    at HTMLScriptElement.script.onload (eval at <anonymous> (iodide.eval-frame.dev.js:93), <anonymous>:83:32)
```

**Comment by mdboom** [comment]
@act65: This is coming from emscripten-created code.  Is this on your locally built copy, one of the official Pyodide releases or at `alpha.iodide.io`?  What version of Chrome?  (I can't reproduce on the 74.0.3729.108 that I have at hand...)

**Comment by mdboom** [comment]
Oh, sorry -- I see you said Chrome 73.  Can you try upgrading to confirm the Chrome version is at issue?

**Comment by act65** [comment]
hmm, my chromebook doesnt seem to want to update any further (will try to figure out...). 
yeah, i think it must to an issue with the version as well. it works fine on my other computer...
traces came from trying to run `https://alpha.iodide.io/notebooks/300/`.

since it just seems like an old unsupported chrome version is the problem should i just close the issue??

**Comment by mdboom** [comment]
Let's keep this open -- it might be something else.  This certainly has worked on that chrome version in the past...

**Comment by osper66** [comment]
For ASUS Cromebooks, current pyodide seems to require more than 32 GB memory.

**Comment by mdboom** [comment]
Hmm...  That's surprising.  What version of Chrome is on the Chromebook?  Can you provide a memory profile from developer tools?  That would show where the memory is going.

**Comment by codingolympics** [comment]
![image](https://user-images.githubusercontent.com/51875201/61918979-2d2e5380-af86-11e9-8cde-0309625ae964.png)
i also get the error,and i found chrome 32 bit version not work.64 bit is ok.

**Comment by donkirkby** [comment]
I'm getting the same error using the [pyodide] web page. Any advice on how to catch this error on [my own site] and display an error message to the user?

Android Chrome version 80.0.3987.149. (Looks current according to the list on [Wikipedia].)

Android 7.1.1

Full error message:

```
pyodide.asm.js:8 Uncaught RangeError: WebAssembly.Memory(): could not allocate memory
    at pyodide.asm.js:8
    at eval (eval at <anonymous> (language-actions.js:56), <anonymous>:376:22)
    at HTMLScriptElement.script.onload (eval at <anonymous> (language-actions.js:56), <anonymous>:83:32)
```

[pyodide]: https://alpha.iodide.io/notebooks/300/
[Wikipedia]: https://en.wikipedia.org/wiki/Google_Chrome_version_history
[my own site]: https://donkirkby.github.io/live-py-plugin/demo/

**Comment by donkirkby** [comment]
It looks like this problem has been fixed in master (7398edb787aba8e229e00a81dbcadb519f5b0baf). I pulled the latest, rebuilt, and deployed. Now my site loads on Android Chrome! I suspect it may be PR #600 that solved the problem.

Thanks for the patch, any plans to tag a new release some time soon?

**Comment by rth** [comment]
> Now my site loads on Android Chrome! I suspect it may be PR #600 that solved the problem.

Thanks for the confirmation. There was the 0.15 release in May which included this fix. 

Closing this issue. Please comment if you still run into difficulties. There is a discussion about Safari specific issues in https://github.com/iodide-project/pyodide/issues/721
